### PR TITLE
Domains: Clean old domain management code

### PR DESCRIPTION
**Pull request by leonardost**
### Changes proposed in this Pull Request

We deployed the new domain management redesigned pages for all users with #59241, but as the new components were mostly based on older components, some obsolete and unused code remained behind. This PR aims to clean that unused code.

### Testing instructions

- No behavior changes should happen because of this PR, but please do a smoke test on the domain management pages (both the site domains and all domains) to ensure everything still looks and works as expected
- Also please test the bulk contact information update page at `/domains/manage?site=all&action=edit-contact-email` and ensure it's working normally

### CODE REVIEW

File: client/my-sites/domains/domain-management/list/index.jsx [hunk 1]
```diff
@@ -1,444 +0,0 @@
-/* eslint-disable wpcalypso/jsx-classname-namespace */
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/list/index.jsx:1]: This component became `SiteDomains`

File: client/my-sites/domains/domain-management/list/wpcom-domain-item.jsx [hunk 2]
```diff
@@ -1,94 +0,0 @@
-import { Button, Gridicon } from '@automattic/components';
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/list/wpcom-domain-item.jsx:1]: This component became `FreeDomainItem`

File: client/my-sites/domains/domain-management/list/list-all.jsx [hunk 3]
```diff
@@ -1,652 +0,0 @@
-import { Button, Card } from '@automattic/components';
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/list/list-all.jsx:1]: This component became `AllDomains`

File: client/my-sites/domains/domain-management/list/list-header.jsx [hunk 4]
```diff
@@ -1,78 +0,0 @@
-import { CompactCard } from '@automattic/components';
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/list/list-header.jsx:1]: This component became `DomainsTableHeader`

File: client/my-sites/domains/domain-management/list/domain-item.jsx [hunk 5]
```diff
@@ -1,474 +0,0 @@
-import { Button, CompactCard } from '@automattic/components';
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/list/domain-item.jsx:1]: This component became `DomainRow`

File: client/my-sites/domains/domain-management/list/all-domains.jsx [hunk 6]
```diff
@@ -763,14 +645,14 @@ class AllDomains extends Component {
...
-				<div className="all-domains__form">{ this.renderActionForm() }</div>
-				<div className="all-domains__container">
+				<div>{ this.renderActionForm() }</div>
+				<div>
 					<QueryAllDomains />
 					<QueryUserPurchases />
 					<Main wideLayout>
 						<SidebarNavigation />
 						<DocumentHead title={ translate( 'Domains', { context: 'A navigation label.' } ) } />
-						<div className="all-domains__items">{ this.renderDomainsList() }</div>
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/list/all-domains.jsx:773]: These classes did not have any associated styling

File: client/my-sites/domains/domain-management/list/all-domains.jsx [hunk 7]
```diff
@@ -260,75 +233,6 @@ class AllDomains extends Component {
...
-
-	getActionResult( domain ) {
-		if ( this.props.isContactEmailEditContext ) {
-			return this.state.contactInfoSaveResults[ domain ] ?? null;
-		}
-
-		return null;
-	}
-
-	renderDomainItem( domain, index ) {
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/list/all-domains.jsx:279]: Not used anymore, this functionality is now done inside `DomainsTable`'s render method, which renders `DomainRow` components

File: client/my-sites/domains/domain-management/list/all-domains.jsx [hunk 8]
```diff
@@ -346,28 +248,6 @@ class AllDomains extends Component {
 		} );
 	};
 
-	renderDomainListHeader() {
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/list/all-domains.jsx:349]: This header is now rendered by `DomainsTableHeader`

File: client/my-sites/domains/domain-management/list/bulk-edit-contact-info.jsx [hunk 9]
```diff
@@ -177,7 +175,7 @@ class BulkEditContactInfo extends Component {
 					onChange={ this.onTransferLockOptOutChange }
 					saveButtonLabel={ translate( 'Save contact info' ) }
 				/>
-				<div className="list__form-buttons">
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/list/bulk-edit-contact-info.jsx:180]: This class didn't have any associated styling

File: client/my-sites/domains/domain-management/list/domains-table-header.jsx [hunk 10]
```diff
@@ -154,7 +154,6 @@ class DomainsTableHeader extends PureComponent {
 			return (
 				<CompactCard className={ listHeaderClasses }>
 					<FormCheckbox
-						className="list__checkbox"
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/list/domains-table-header.jsx:157]: There was no associated styling for this class

File: client/my-sites/domains/domain-management/list/style.scss [hunk 11]
```diff
@@ -28,36 +28,10 @@
 	}
 }
 
-.domain-management-list__notice {
-	margin-bottom: 0;
-}
-
-.domain-management-list-item {
-	&.busy {
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/list/style.scss:36]: This `.domain-management-list-item` class with a `.busy` modifier is not used anymore

File: client/my-sites/domains/domain-management/list/style.scss [hunk 12]
```diff
@@ -111,193 +85,6 @@
 	}
 }
 
-.domain-item,
-.list-header {
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/list/style.scss:115]: Both `DomainItem` and `ListHeader` were removed, so these classes are unused and can be removed too. `domain-item` CSS classes are now `domain-row` classes in `domain-row.scss`

File: client/my-sites/domains/domain-management/list/style.scss [hunk 13]
```diff
@@ -111,193 +85,6 @@
...
-	border-left: 3px solid var( --studio-orange-50 );
-}
-
-.domain-item__status-alert {
-	border-left: 3px solid var( --color-error );
-}
-
-.domain-item__status-premium {
-	border-left: 3px solid var( --color-premium-domain );
-}
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/list/style.scss:229]: These classes are now in `domain-row.scss`

**Review by gius80 — APPROVED** [review]
LGTM! I tested all domains, site domains and email bulk edit - on mobile and desktop - and everything works as expected. Thanks!

**Pull request merged**